components/system-api/src/public.rs

```diff
@@ -97,7 +97,7 @@ impl SystemApi {
 
     fn handle_stats_info(self, name: &'static str, api_scope: &mut ApiScope) -> Self {
         let self_ = self.clone();
-        api_scope.endpoint(name, move |_query: ()| {
+        api_scope.endpoint(name, move |_query: ()| -> api::Result<_> {
```

Comment:
Nit: Specifying the return type is redundant here and in other places in this file.